app/src/main/java/org/stepic/droid/code/ui/CodeAnalyzer.kt

```diff
@@ -0,0 +1,104 @@
+package org.stepic.droid.code.ui
+
+import org.stepic.droid.util.countWhile
+import org.stepic.droid.util.substringOrNull
+
+/**
+ * Class for smart code analyzing
+ */
+object CodeAnalyzer {
+    private const val LINE_BREAK = '\n'
+
+    private val brackets = hashMapOf(
+        "{" to "}",
+        "(" to ")",
+        "[" to "]"
+    )
+
+    private val indentation = hashMapOf( // code specific indentation symbols
+            "py" to hashSetOf(
+                    ":"
+            )
+    )
+
+    private val quotes = hashSetOf(
+        "\"", "'"
+    )
+
+    private fun getIndentForCurrentLine(start: Int, string: String) : Int {
```

Comment:
is `start` current cursor position? Let's rename it or add some comment

Reply:
we can use `line` instead of `string` name

Reply:
We can't use `line` because whole string should be passed in this method or at least string that contains current line and previous